Gems/Atom/RHI/Code/Source/RHI.Private/FactoryManagerSystemComponent.cpp

```diff
@@ -149,6 +150,18 @@ namespace AZ
             // If there's more that one factory registered then we need to decide which one to use.
             if (m_registeredFactories.size() > 1)
             {
+                // Load factories priority list from the settings registry
+                if (auto* registry = AZ::SettingsRegistry::Get(); registry != nullptr)
+                {
+                    m_factoriesPriority.clear();
+                    if (registry->GetObject(m_factoriesPriority, "/O3DE/Atom/RHI/FactoryManager/factoriesPriority"))
```

Comment:
The clear probably wants to be inside the `if (GetObject)` block, that way we don't clear it if the registry key is missing.

Reply:
If we do that, we'll clear the data that we've just read. The idea was that if the registry key is missing the default priority is used. An empty `m_factoriesPriority` ensures this case

Reply:
We can use this logic to see if this key exists before c=doing a clear
using Type = AZ::SettingsRegistryInterface::Type;
if (settingsRegistry.GetType(<key>) == Type::NoType)
{
}